federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPOperationManager.java

```diff
@@ -370,26 +370,36 @@ private SearchControls getSearchControls(Collection<String> returningAttributes,
     }
 
     public String getFilterById(String id) {
-        String filter = null;
+        StringBuilder filter = new StringBuilder();
+        filter.insert(0, "(&");
 
         if (this.config.isObjectGUID()) {
             byte[] objectGUID = LDAPUtil.encodeObjectGUID(id);
-
-            filter = "(&(objectClass=*)(" + getUuidAttributeName() + LDAPConstants.EQUAL + LDAPUtil.convertObjectGUIDToByteString(objectGUID) + "))";
+            filter.append("(objectClass=*)(").append(
+                    getUuidAttributeName()).append(LDAPConstants.EQUAL)
+                .append(LDAPUtil.convertObjectGUIDToByteString(
+                    objectGUID)).append(")");
 
         } else if (this.config.isEdirectoryGUID()) {
-            filter = "(&(objectClass=*)(" + getUuidAttributeName().toUpperCase() + LDAPConstants.EQUAL + LDAPUtil.convertGUIDToEdirectoryHexString(id) + "))";
+            filter.append("(objectClass=*)(").append(getUuidAttributeName().toUpperCase())
+                .append(LDAPConstants.EQUAL
+                ).append(LDAPUtil.convertGUIDToEdirectoryHexString(id)).append(")");
+        } else {
+            filter.append("(objectClass=*)(").append(getUuidAttributeName()).append(LDAPConstants.EQUAL)
+                .append(id).append(")");
         }
 
-        if (filter == null) {
-            filter = "(&(objectClass=*)(" + getUuidAttributeName() + LDAPConstants.EQUAL + id + "))";
+        if (config.getCustomUserSearchFilter() != null) {
+            filter.append(config.getCustomUserSearchFilter());
         }
 
         if (logger.isTraceEnabled()) {
-            logger.tracef("Using filter for lookup user by LDAP ID: %s", filter);
+            logger.tracef("Using filter for lookup user by LDAP ID: %s", filter.toString());
```

Comment:
Minor note: The "trace" message displayed here won't contain the last parentheses, which you added on the line 400. So it seems it will be better to change the order and make sure that the `filter.append('"')` is called first and then the `trace` is called afterwards.